controller/src/main/java/io/pravega/controller/store/stream/PersistentStreamBase.java

```diff
@@ -1931,8 +1931,9 @@ boolean compareWriterPositions(Map<Long, Long> position1, Map<Long, Long> positi
                 () -> getTransactionRecords(epoch, txnIds.subList(from.get(), till.get())).thenAccept(txns -> {
             for (int i = 0; i < txns.size(); i++) {
                 ActiveTxnRecord txnRecord = txns.get(i);
```

Comment:
when will `txnRecord.commitOrder` be different from` orders.get(index)`?

Reply:
Because we first create znode and then we update the txn record, we could have failed after 1st. Or could have recreated another znode in a different call that then doesn’t update the record.